src/pages/Version.tsx

```diff
@@ -0,0 +1,154 @@
+import { useState, useEffect } from "react";
+import { useLazyQuery } from "@apollo/client";
+import { useParams, Redirect } from "react-router-dom";
+import { BreadCrumb } from "components/Breadcrumb";
+import { PatchAndTaskFullPageLoad } from "components/Loading/PatchAndTaskFullPageLoad";
+import { PageTitle } from "components/PageTitle";
+import { PatchStatusBadge } from "components/PatchStatusBadge";
+import {
+  PageWrapper,
+  PageContent,
+  PageLayout,
+  PageSider,
+} from "components/styles";
+import { pollInterval } from "constants/index";
+import { commitQueueAlias } from "constants/patch";
+import { getCommitQueueRoute, getPatchRoute } from "constants/routes";
+import { useToastContext } from "context/toast";
+import {
+  VersionQuery,
+  VersionQueryVariables,
+  IsPatchConfigurableQuery,
+  IsPatchConfigurableQueryVariables,
+} from "gql/generated/types";
+import { GET_VERSION, GET_IS_PATCH_CONFIGURED } from "gql/queries";
+import { usePageTitle, useNetworkStatus } from "hooks";
+import { PageDoesNotExist } from "pages/404";
+import { validatePatchId } from "utils/validators";
+import { BuildVariants } from "./version/BuildVariants";
+import { ActionButtons } from "./version/index";
+import { Metadata } from "./version/Metadata";
+import { PatchTabs } from "./version/PatchTabs";
+
+export const VersionPage: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+
+  const dispatchToast = useToastContext();
+
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const [redirectURL, setRedirectURL] = useState("");
+  const [isLoadingData, setIsLoadingData] = useState(true);
+
+  const [getPatch, { data: patchData }] = useLazyQuery<
+    IsPatchConfigurableQuery,
+    IsPatchConfigurableQueryVariables
+  >(GET_IS_PATCH_CONFIGURED, {
+    variables: {
+      id,
+    },
+  });
+
+  const [getVersion, { data, error, startPolling, stopPolling }] = useLazyQuery<
+    VersionQuery,
+    VersionQueryVariables
+  >(GET_VERSION, {
+    variables: { id },
+    pollInterval,
+    onError: (e) =>
+      dispatchToast.error(`There was an error loading the patch: ${e.message}`),
+    onCompleted() {
+      setIsLoadingData(false);
+    },
+  });
+
+  useNetworkStatus(startPolling, stopPolling);
+
+  // If we are viewing a patch we should first check if its configured before trying to check
```

Comment:
I feel like this comment isn't quite accurate; Doesn't validatePatchId just check if it's a patch or not a patch? I don't see where we're checking whether or not it's a configured patch. (Edit: does this happen at line 90? In which case can we make this comment more helpful?)